skxray/fitting/xrf_model.py

```diff
@@ -0,0 +1,1042 @@
+    fwhm_fanoprime : float
+        global fitting parameter for peak width
+    e_offset : float
+        offset of energy calibration
+    e_linear : float
+        linear coefficient in energy calibration
+    e_quadratic : float
+        quadratic coefficient in energy calibration
+
+    Returns
+    -------
+    array:
+        gaussian peak profile
+    """
+    def get_sigma(center):
+        temp_val = 2 * np.sqrt(2 * np.log(2))
+        return np.sqrt((fwhm_offset/temp_val)**2 + center*epsilon*fwhm_fanoprime)
+
+    x = e_offset + x * e_linear + x**2 * e_quadratic
+
+    return gaussian(x, area, center+delta_center,
+                    delta_sigma+get_sigma(center)) * ratio * ratio_adjust
+
+
+class ElementModel(Model):
+
+    __doc__ = _gen_class_docs(element_peak_xrf)
+
+    def __init__(self, *args, **kwargs):
+        super(ElementModel, self).__init__(element_peak_xrf, *args, **kwargs)
+        self.set_param_hint('epsilon', value=2.96, vary=False)
+
+
+def _set_parameter_hint(param_name, input_dict, input_model,
+                        log_option=False):
+    """
+    Set parameter hint information to lmfit model from input dict.
+    .. warning :: This function mutates the input values.
+
+    Parameters
+    ----------
+    param_name : str
+        parameter used for fitting
+    input_dict : dict
+        all the initial values and constraints for given parameters
+    input_model : object
+        model object used in lmfit
+    log_option : bool
+        option for logger
+    """
+    if input_dict['bound_type'] == 'none':
+        input_model.set_param_hint(name=param_name, value=input_dict['value'], vary=True)
+    elif input_dict['bound_type'] == 'fixed':
+        input_model.set_param_hint(name=param_name, value=input_dict['value'], vary=False)
+    elif input_dict['bound_type'] == 'lohi':
+        input_model.set_param_hint(name=param_name, value=input_dict['value'], vary=True,
+                                   min=input_dict['min'], max=input_dict['max'])
+    elif input_dict['bound_type'] == 'lo':
+        input_model.set_param_hint(name=param_name, value=input_dict['value'], vary=True,
+                                   min=input_dict['min'])
+    elif input_dict['bound_type'] == 'hi':
+        input_model.set_param_hint(name=param_name, value=input_dict['value'], vary=True,
+                                   min=input_dict['max'])
+    else:
+        raise ValueError("could not set values for {0}".format(param_name))
+    if log_option:
+        logger.debug(' {0} bound type: {1}, value: {2}, range: {3}'.
+                    format(param_name, input_dict['bound_type'], input_dict['value'],
+                           [input_dict['min'], input_dict['max']]))
+
+
+def update_parameter_dict(xrf_parameter, fit_results):
+    """
+    Update fitting parameters dictionary according to given fitting results,
+    usually obtained from previous run.
+    .. warning :: This function mutates the input values.
```

Comment:
@tacaswell thinks this might need a whitespace line above it.  If you build the docs with Sphinx, you should see a bright red box with this warning.